### DISCUSSION TITLE
Using English Models = "ValueError: This model doesn't have language tokens so it can't perform lang id"

### DISCUSSION THREAD

**Discussion by ThioJoe** [openai/whisper#17] [Q&A] [answered]
I'm using the shorter code example from the readme - the first one under "Python usage".

It works except if I try to set the model to any of the ".en" (English only) ones, in which case I get the error:

`ValueError: This model doesn't have language tokens so it can't perform lang id`

Using the english specific model does work when using whisper via command line. Does something more need to be done if running it in a script?

Here's the full traceback:

```
Traceback (most recent call last):
  File "D:\whateverpath\whisper\myscript.py", line 10, in <module>
    result = model.transcribe("audio.mp3")
  File "D:\whateverpath\whisper\transcribe.py", line 82, in transcribe
    _, probs = model.detect_language(segment)
  File "C:\Users\Joe\miniconda3\envs\whisper\lib\site-packages\torch\autograd\grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "D:\whateverpath\whisper\decoding.py", line 35, in detect_language
    raise ValueError(f"This model doesn't have language tokens so it can't perform lang id")
ValueError: This model doesn't have language tokens so it can't perform lang id
```

**Comment by astrowonk** [comment, marked as answer]
Try `model.transcribe("file.wav",language='english')`

Using `small.en` only slightly faster than `small` on my M1 Mac Mini, and actually slightly less accurate on my one self-recorded example audio file.

**Reply by ThioJoe** [reply]
Yep that worked

**Comment by jayavanth** [comment]
If you're using `model.decode`, pass it to `DecodingOptions`

`options = whisper.DecodingOptions(language='english')`

**Comment by OrbitPeppermint** [comment]
We had the same problem, fixed it by updating whisper to the latest Version with 
`` pip install -U openai-whisper ``
